pkg/crane/digest.go

```diff
@@ -19,6 +19,9 @@ func Digest(ref string, opt ...Option) (string, error) {
 	o := makeOptions(opt...)
 	if o.platform != nil {
 		desc, err := getManifest(ref, opt...)
+		if !desc.MediaType.IsIndex() {
```

Comment:
It looks like desc will be `nil` if there are errors fetching the manifest (e.g., HTTP 404, 403), which will result in a nil panic at this line. Should this check happen after the err check?

Reply:
Yes this should be after the err check. I moved it.